Coreboot Toolchain fails to Build (gnat broken?)

### Steps To Reproduce

Steps to reproduce the behavior:
1. `nix-build -A pkgs.coreboot-toolchain.i386`

This will result in `pkgs.gnat6` to fail to build.

### Build log

```
...
make[3]: Entering directory '/build/build/libdecnumber'
source='../../gcc-6.5.0/libdecnumber/decNumber.c' object='decNumber.o' libtool=no gcc  -I../../gcc-6.5.0/libdecnumber -I.  -g -W -Wall -Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition -Wmissing-format-attribute -Wcast-qual -pedantic -Wno-long-long  -I../../gcc-6.5.0/libdecnumber -I.   -c ../../gcc-6.5.0/libdecnumber/decNumber.c
In file included from /nix/store/a0s1vq35j3g6aa757v7yg0a001l01ng4-glibc-2.34-210-dev/include/string.h:519,
                 from ../../gcc-6.5.0/libdecnumber/decNumber.c:177:
/nix/store/a0s1vq35j3g6aa757v7yg0a001l01ng4-glibc-2.34-210-dev/include/bits/string_fortified.h:110:50: error: macro "__fortified_attr_access" requires 3 arguments, but only 1 given
In file included from /nix/store/a0s1vq35j3g6aa757v7yg0a001l01ng4-glibc-2.34-210-dev/include/string.h:519,
                 from ../../gcc-6.5.0/libdecnumber/decNumber.c:177:
/nix/store/a0s1vq35j3g6aa757v7yg0a001l01ng4-glibc-2.34-210-dev/include/bits/string_fortified.h:110: error: expected ',' or ';' before '__fortified_attr_access'
make[3]: *** [Makefile:191: decNumber.o] Error 1
make[3]: Leaving directory '/build/build/libdecnumber'
make[2]: *** [Makefile:8706: all-stage1-libdecnumber] Error 2
make[2]: Leaving directory '/build/build'
make[1]: *** [Makefile:21252: stage1-bubble] Error 2
make[1]: Leaving directory '/build/build'
make: *** [Makefile:21573: bootstrap] Error 2
error: builder for '/nix/store/983dhfb06fds2x1yp8zi060ybqgbshmf-gnat-6.5.0.drv' failed with exit code 2;
...
```

Full log: https://gist.github.com/blitz/ee604e1ddf51c9c8e91fe3c5edb36032

### Additional context

I'm currently bisecting this, but it will take a bit. First bad commit is 55c4a726fe944d4a637affde93b2594f0bc0aa07, last good is 5ee3e9f434e1cb1263c331e95d0b3d3bb384e3ec.

### Notify maintainers

@felixsinger 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.9, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4`
 - channels(root): `"nixos-21.11.337659.06db2e21974, nixos-hardware, nixos-unstable-22.11pre380124.5ce6597eca7, tuxedo"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

This was fixed, but now coreboot-toolchain is broken again due to a broken gnat11 build.

```
$ /nix/store/a4dvmfc4aqvviq78ixg7kydyy6sj5fmk-gnat-11.3.0.drv
...
gnatbind -nostdinc -I- -I. -Iada/generated -Iada -Iada/gcc-interface -I../../gcc-11.3.0/gcc/ada -I../../gcc-11.3.0/gcc/ada/gcc-interface -Iada/libgnat -I../../gcc-11.3.0/gcc/ada/libgnat -o b_gnat1.adb -n ada/gnat1drv.ali
g++ -std=c++11   -g -DIN_GCC     -fno-exceptions -fno-rtti -fasynchronous-unwind-tables -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wno-error=format-diag -Wno-format -Wmissing-format-attribute -Woverloaded-virtual -pedantic -Wno-long-long -Wno-variadic-macros -Wno-overlength-strings -fno-common  -DHAVE_CONFIG_H  -DGENERATOR_FILE -static-libstdc++ -static-libgcc  -o build/genpreds \
    build/genpreds.o build/rtl.o build/read-rtl.o build/ggc-none.o build/vec.o build/min-insn-modes.o build/gensupport.o build/print-rtl.o build/hash-table.o build/sort.o build/read-md.o build/errors.o ../build-x86_64-unknown-linux-gnu/libiberty/libiberty.a
g++ -std=c++11   -g -DIN_GCC     -fno-exceptions -fno-rtti -fasynchronous-unwind-tables -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wno-error=format-diag -Wno-format -Wmissing-format-attribute -Woverloaded-virtual -pedantic -Wno-long-long -Wno-variadic-macros -Wno-overlength-strings -fno-common  -DHAVE_CONFIG_H  -DGENERATOR_FILE -static-libstdc++ -static-libgcc  -o build/genflags \
    build/genflags.o build/rtl.o build/read-rtl.o build/ggc-none.o build/vec.o build/min-insn-modes.o build/gensupport.o build/print-rtl.o build/hash-table.o build/sort.o build/read-md.o build/errors.o ../build-x86_64-unknown-linux-gnu/libiberty/libiberty.a
error: "s-aotase.adb" must be compiled
error: ("/nix/store/wq6f3cxqzip6gv6c9s4za9z48af5sxj8-gnatboot-12.1.0-2/lib/gcc/x86_64-pc-linux-gnu/12.1.0/adalib/s-aotase.ali" is obsolete and read-only)
error: "s-atoope.ads" must be compiled
error: ("/nix/store/wq6f3cxqzip6gv6c9s4za9z48af5sxj8-gnatboot-12.1.0-2/lib/gcc/x86_64-pc-linux-gnu/12.1.0/adalib/s-atoope.ali" is obsolete and read-only)
error: "s-atopri.adb" must be compiled
error: ("/nix/store/wq6f3cxqzip6gv6c9s4za9z48af5sxj8-gnatboot-12.1.0-2/lib/gcc/x86_64-pc-linux-gnu/12.1.0/adalib/s-atopri.ali" is obsolete and read-only)
error: "i-c.adb" must be compiled
error: ("/nix/store/wq6f3cxqzip6gv6c9s4za9z48af5sxj8-gnatboot-12.1.0-2/lib/gcc/x86_64-pc-linux-gnu/12.1.0/adalib/i-c.ali" is obsolete and read-only)
make[3]: *** [../../gcc-11.3.0/gcc/ada/gcc-interface/Make-lang.in:1014: ada/b_gnat1.adb] Error 5
make[3]: *** Waiting for unfinished jobs....
rm gcc.pod
make[3]: Leaving directory '/build/build/gcc'
make[2]: *** [Makefile:4791: all-stage1-gcc] Error 2
make[2]: Leaving directory '/build/build'
make[1]: *** [Makefile:24626: stage1-bubble] Error 2
make[1]: Leaving directory '/build/build'
make: *** [Makefile:24963: bootstrap] Error 2
```

--- a/pkgs/development/tools/misc/coreboot-toolchain/default.nix
+++ b/pkgs/development/tools/misc/coreboot-toolchain/default.nix
@@ -40,7 +40,7 @@ let
       dontInstall = true;
 
       postPatch = ''
-        patchShebangs $out/util/crossgcc/buildgcc
+        patchShebangs util/crossgcc/buildgcc
 
         mkdir -p util/crossgcc/tarballs
 

--- a/pkgs/development/tools/misc/coreboot-toolchain/default.nix
+++ b/pkgs/development/tools/misc/coreboot-toolchain/default.nix
@@ -9,10 +9,12 @@ let
     , getopt
     , git
     , gnat11
+    , gcc
     , lib
     , perl
     , stdenvNoCC
     , zlib
+    , withAda ? true
     }:
 
     stdenvNoCC.mkDerivation rec {
@@ -33,7 +35,7 @@ let
       };
 
       nativeBuildInputs = [ bison curl git perl ];
-      buildInputs = [ flex gnat11 zlib ];
+      buildInputs = [ flex zlib (if withAda then gnat11 else gcc) ];
 
       enableParallelBuilding = true;
       dontConfigure = true;

